src/overlays/actors/ovl_Obj_Um/z_obj_um.h

```diff
@@ -2,18 +2,82 @@
 #define Z_OBJ_UM_H
 
 #include "global.h"
+#include "objects/object_um/object_um.h"
+#include "overlays/actors/ovl_En_Horse/z_en_horse.h"
 
 struct ObjUm;
 
 typedef void (*ObjUmActionFunc)(struct ObjUm*, PlayState*);
 
 typedef struct ObjUm {
-    /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x18];
-    /* 0x015C */ ObjUmActionFunc actionFunc;
-    /* 0x0160 */ char unk_160[0x384];
+    /* 0x000 */ DynaPolyActor dyna;
+    /* 0x15C */ ObjUmActionFunc actionFunc;
+    /* 0x160 */ SkelAnime skelAnime;
+    /* 0x1A4 */ Vec3s jointTable[UM_LIMB_MAX];
+    /* 0x228 */ Vec3s morphTable[UM_LIMB_MAX];
+    /* 0x2AC */ s16 wheelRot;
+    /* 0x2AE */ s16 type;
+    /* 0x2B0 */ s16 initialPathIdx;
+    /* 0x2B4 */ s32 unk_2B4;
+    /* 0x2B8 */ EnHorse* donkey;
+    /* 0x2BC */ s32 pathIdx;
+    /* 0x2BE */ s32 pointIdx;
+    /* 0x2C4 */ Vec3f unk_2C4;
+    /* 0x2D0 */ Vec3f unk_2D0;
+    /* 0x2DC */ Vec3f unk_2DC;
+    /* 0x2E8 */ Vec3f unk_2E8;
+    /* 0x2F4 */ s32 flags;
+    /* 0x2BC */ Vec3s unk_2F8;
+    /* 0x2FE */ Vec3s unk_2FE;
+    /* 0x304 */ s32 unk_304;
+    /* 0x308 */ Vec3f unk_308;
+    /* 0x314 */ s32 potsLife[3];
+    /* 0x320 */ s32 wasPotHit[3]; // resets to false in the same frame
+    /* 0x32C */ Vec3f potPos[3];
+    /* 0x350 */ s32 unk_350; // unused counter?
+    /* 0x354 */ s32 unk_354; // unused?
+    /* 0x358 */ EnHorse* bandit1;
+    /* 0x35C */ EnHorse* bandit2;
+    /* 0x360 */ Vec3f unk_360[16];
+    /* 0x420 */ s32 unk_420; // ?
+    /* 0x424 */ ColliderCylinder banditsCollisions[2];
+    /* 0x4BC */ Vec3f cartBedPos;
+    /* 0x4C8 */ u16 lastTime;
+    /* 0x4CC */ s32 unk_4CC;
+    /* 0x4D0 */ s32 eyeTexIndex;
+    /* 0x4D4 */ s32 mouthTexIndex;
+    /* 0x4D8 */ s32 unk_4D8;
+    /* 0x4DC */ s32 unk_4DC;
+    /* 0x4E0 */ s32 arePotsBroken;
 } ObjUm; // size = 0x4E4
 
 extern const ActorInit Obj_Um_InitVars;
 
+typedef enum {
+    /* 0 */ OBJ_UM_TYPE_TERMINA_FIELD,
+    /* 1 */ OBJ_UM_TYPE_RANCH,
+    /* 2 */ OBJ_UM_TYPE_PRE_MILK_RUN, // milk road, pre-minigame
+    /* 3 */ OBJ_UM_TYPE_MILK_RUN_MINIGAME,
+    /* 4 */ OBJ_UM_TYPE_POST_MILK_RUN // milk road, post-minigame
+} ObjUmType;
+
+#define OBJ_UM_PARSE_TYPE(params) (((params) & 0xFF00) >> 8)
+#define OBJ_UM_PARSE_PATH_IDX(params) ((params) & 0xFF)
+
+#define OBJ_UM_FLAG_0000 0x0000
+#define OBJ_UM_FLAG_0001 0x0001
```

Comment:
```suggestion
#define OBJ_UM_FLAG_0000 0
#define OBJ_UM_FLAG_0001 (1 << 0)
```
Same with the rest as well please :)